src/test/lib/lit-extended_test.ts

```diff
@@ -203,6 +203,36 @@ suite('lit-extended', () => {
       assert.equal(count2, 1);
     });
 
+    test('allows updating event listener without extra calls to remove/addEventListener', () => {
+      let listener: Function|null;
+      const t = () => html`<div on-click=${listener}></div>`;
+      render(t(), container);
+      const div = container.firstChild as HTMLElement;
+      let addCount = 0;
+      let removeCount = 0;
+      div.addEventListener = () => addCount++;
+      div.removeEventListener = () => removeCount++;
+      listener = () => {};
```

Comment:
Since we're interceding the handler, it'd be good to test that the listener is called.

Reply:
There's [another test](https://github.com/Polymer/lit-html/blob/fix-306/src/test/lib/lit-extended_test.ts#L188) for changing the event listener. This one was just to ensure that changing it doesn't spam add/removeEventListener. wdyt?